### ISSUE TITLE
[Feature request] Lock phone when proximity sensor is covered

### ISSUE THREAD

**Issue by bishtawi** [wesaphzt/privatelock#24]
Hi,

Big fan of this app. It is great for security, so thank you. One feature that I think fits in the spirit of the app is locking the phone when the proximity sensor is covered for at least a second (duration configurable ideally).

**Comment by beerisgood** [comment]
You mean like the app WaveUp ?

**Comment by wesaphzt** [comment]
Thank you @bishtawi :) I use this as a daily driver on my phone so I'm always looking to improve the app.
It's a good idea, but WaveUp implements this as @beerisgood says. I use WaveUp myself but just for unlocking the phone.
Saying that though, it could be worth putting in so as not to have to use two apps.

**Comment by bishtawi** [comment]
Yup, looks like WaveUp has the exact feature I am looking for. I have no need for WaveUp's primary functionality and like you said, I'd rather not have to use two apps. Locking the phone when the proximity sensor is covered seems like an appropriate feature for PrivateLock to adopt as it fits in the spirit of the app IMO. I'll use WaveUp in the meantime.

**Comment by Chefche** [comment]
Hello:
Thanks for this software.
I can only guess that the feature with the proximity sensor is implemented. Since on the other hand, locking always requires the PIN as stated elsewhere, it would be helpful if the proximity sensor feature would be switchable.
Thanks and best regards
